app/javascript/mastodon/features/emoji/emoji.js

```diff
@@ -36,6 +36,11 @@ const emojify = (str, customEmojis = {}) => {
     } else if (tag >= 0) { // <, &
       rend = str.indexOf('>;'[tag], i + 1) + 1;
       if (!rend) break;
+      if (str.slice(i, rend) === '<span class="invisible">') {
+        // avoid emojifying on invisible text
+        const span = '</span>';
```

Comment:
this wont work with nested spans correctly. not sure how much we care about that?

Reply:
Yes, I understand that It's safer to handle by whole anchor text. But with that method, the emoji domain will not be comverted...

Reply:
alt way (visible parts also doesn't be converted)
```
if (str.slice(i, rend).startsWith('<a ')) {
  const a = '</a>';
  rend = str.indexOf(a, rend) + a.length;
}
```